### PR TITLE
Add Content switcher and Page menu buttons

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Related to #485 by @zackkrida

## Description

This PR adds the buttons for the Content Switcher (the button that allows the user to select the content type to search) and the Page menu (with links to content pages on Openverse). It also updates the button styles to use `tertiary` for the button that has a border, and `action-menu` for the button that has no border. Another change is to set the `pressed` style based on the `aria-pressed` or `aria-expanded` properties.
This change was extracted from a larger Content Switcher [draft PR](https://github.com/WordPress/openverse-frontend/pull/490) based on Krysal's [comment](https://github.com/WordPress/openverse-frontend/pull/490#issuecomment-1000562201) in a larger Content Switcher [draft PR](https://github.com/WordPress/openverse-frontend/pull/490).

To prevent the need for props drilling, it also uses `provide` in the VHeader component to provide all child components with values of `sHeaderScrolled` and `isMdScreen`.

## Testing Instructions

Run the project using `pnpm run dev`, and go to `localhost:8443/search/images/?q=cat`. You should see two buttons in the header.
#### Desktop, header not scrolled
Screen Shot 2022-01-03 at 9 04 02 PM

#### Desktop, header scrolled
Screen Shot 2022-01-03 at 9 03 50 PM

#### Mobile, header not scrolled
Screen Shot 2022-01-03 at 9 06 57 PM

#### Mobile, header scrolled
Screen Shot 2022-01-03 at 9 03 31 PM

Please note, that the buttons are not functioning yet, and their placement is not correct.
## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by zackkrida** [comment]
> Please note, that the buttons are not functioning yet, and their placement is not correct.

👍 Visually, they look nice so far!

File: src/assets/icons/audio-content.svg [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<svg viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
+  <g id="icon">
+    <path fill-rule="evenodd" clip-rule="evenodd" d="M3.25 7L3.25 17H4.75V7H3.25Z" fill="currentColor"/>
+    <path fill-rule="evenodd" clip-rule="evenodd" d="M19.25 7V17H20.75V7H19.25Z" fill="currentColor"/>
```
Comment (krysal) [inline src/assets/icons/audio-content.svg:4]: **small suggestion:** The `fill="currentColor"` could be indicated on the `g` tag and that way it is not necessary to repeat it on every `path`.

**Review by krysal — COMMENTED** [review]
This PR looks more manageable, thanks! Do you want us to check something specifically here? I got an error when I try to see the VContentSwitcherButton on Storybook.

**Comment by zackkrida** [comment]
@obulat is this meant to be a draft PR, or would you like to merge these buttons and then implement functionality in a subsequent PR? Also, will #490 be closed eventually or do you plan to revive it?

**Comment by obulat** [comment]
It's a draft PR not because the buttons are not ready, but because they need other components (popover, modal) to be useful.

**Comment by zackkrida** [comment]
Okay, I've switched this to a draft PR for now.

**Comment by obulat** [comment]
Closing this PR in favor of #607.

**Pull request closed without merging**